File: 3170-lexicographically-minimum-string-after-removing-stars/3170-lexicographically-minimum-string-after-removing-stars.cpp
class Solution {
public:
    string clearStars(string s) {
        vector<int> positions[26];
        // map<int, bool> deleted;
        vector<bool> deleted(s.size(), false);
        
        for(int i=0; i<s.size(); i++) {
            if(s[i] != '*') {
                positions[s[i]-'a'].push_back(i);
            }
            else {
                for(int j=0; j<26; j++) {
                    if(positions[j].size() > 0) {
                        deleted[positions[j].back()] = true;
                        deleted[i] = true;
                        positions[j].pop_back();
                        break;
                    }
                }
            }
        }
        
        string res="";
        for(int i=0; i<s.size(); i++) {
            if (deleted[i] == false) {
                res.push_back(s[i]);
            }
        }
    return res;
    }
};

File: 3170-lexicographically-minimum-string-after-removing-stars/3170-lexicographically-minimum-string-after-removing-stars.py
class Solution(object):
    def clearStars(self, s):
        """
        :type s: str
        :rtype: str
        """
        deleted = [False for _ in range(len(s))]
        positions = [[] for _ in range(26)]
        
        for i in range(len(s)):
            if s[i] != '*':
                positions[ord(s[i]) - ord('a')].append(i)
            else:
                for j in range(26):
                    if len(positions[j]) > 0:
                        deleted[positions[j][-1]] = True
                        deleted[i] = True
                        positions[j].pop()
                        break
                    
        
        res = ""
        for i in range(len(s)):
            # if i not in deleted.keys():
                # res += s[i]
            if deleted[i] == False:
                res += s[i]
                
        # print(res)
        return res
                
